### PR TITLE
Fix weak spacing broken in math

### PR DESCRIPTION
**Pull request by sitandr**
A small fix of `dif` not having a spacing in math (in equations like `f(x) dif x`). That problem was introduced in #1633, since `dif` itself wasn't tested.

The problem emerges from nested sequences used for representing math, like `sequence(f, (x), sequence(thin space, d), x)`, so space in inner sequence is cut because it is weakness at start of sequence. I fixed it with replacing iterating over sequence elements with iterating over nested sequence and added tests to check it works.

### CODE REVIEW

File: crates/typst/src/model/content.rs [hunk 1]
```diff
@@ -91,6 +91,18 @@ impl Content {
...
     }
 
+    /// Also auto expands sequence of sequences into flat sequence
+    pub fn to_sequence_recursive(&self) -> Option<Box<dyn Iterator<Item = &Self> + '_>> {
+        if !self.is::<SequenceElem>() {
+            return None;
+        }
+        Some(Box::new(
+            self.attrs.iter().filter_map(Attr::child).flat_map(|e| {
+                e.to_sequence_recursive().unwrap_or(Box::new(iter::once(e)))
```
Comment (laurmaedje) [inline crates/typst/src/model/content.rs:101]: especially since this allocates once _per element_ and not per sequence

Comment (sitandr) [reply crates/typst/src/model/content.rs:101]: Oh, sorry a lot. That was quite stupid, I don't know what messed with my mind.

**Comment by sitandr** [comment]
I'm sorry for the delay. I initially thought that `resolve` is like "okay, it's fine, I understand it and will try to fix it soon". So now I guess it was a mistake, thanks!

**Comment by laurmaedje** [comment]
No problem! Maybe that's just the way I interpret "resolve" :)

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**